File: src/main/java/uk/gov/ukho/battleshipsboot/controllers/ShipController.java
package uk.gov.ukho.battleshipsboot.controllers;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.RequestEntity;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import uk.gov.ukho.battleshipsboot.facades.ShipPlacingFacade;
import uk.gov.ukho.battleshipsboot.model.game.BoardPosition;
import uk.gov.ukho.battleshipsboot.model.game.Orientation;
import uk.gov.ukho.battleshipsboot.model.ships.Destroyer;
import uk.gov.ukho.battleshipsboot.model.ships.Ship;


@RestController
public class ShipController {

    @Autowired
    private ShipPlacingFacade shipPlacingFacade;

    @RequestMapping(value = "/placedestoyer", method = RequestMethod.POST)
    public ResponseEntity<ShipPlaceRequestWrapper> placeDestroyer(
            @RequestBody ShipPlaceRequestWrapper shipPlaceRequestWrapper)
    {
        Destroyer destoyer = new Destroyer(shipPlaceRequestWrapper.getOrientation()
                , shipPlaceRequestWrapper.getBoardPosition());
        shipPlacingFacade.placeShip(shipPlaceRequestWrapper.getPlayerId()
                , shipPlaceRequestWrapper.getGameId(), destoyer);
        System.out.println("End point hit!");
        return new ResponseEntity<>(shipPlaceRequestWrapper,HttpStatus.CREATED);
    }

}

File: src/main/java/com/harragan/battleshipsboot/controllers/ShipController.kt
package com.harragan.battleshipsboot.controllers

import com.harragan.battleshipsboot.facades.ShipPlacingFacade
import com.harragan.battleshipsboot.model.kotlinmodel.ships.Ship

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*


@RestController
class ShipController(private val shipPlacingFacade: ShipPlacingFacade) {
    @ResponseStatus(HttpStatus.CREATED)
    @RequestMapping(value = ["/placeShip"], method = [RequestMethod.POST])
    fun placeShip(
            @RequestBody shipPlaceRequestWrapper: ShipPlaceRequestWrapper): Ship {
        val ship = Ship(orientation = shipPlaceRequestWrapper.orientation, boardPosition = shipPlaceRequestWrapper.boardPosition, type = shipPlaceRequestWrapper.shipType)
        shipPlacingFacade.placeShip(shipPlaceRequestWrapper.playerId, shipPlaceRequestWrapper.gameId, ship)
        return ship
    }
}


